lib/awful/tag.lua

```diff
@@ -1355,6 +1355,32 @@ capi.screen.add_signal("tag::history::update")
 
 capi.screen.connect_signal("tag::history::update", tag.history.update)
 
+capi.screen.connect_signal("removed", function(s)
+    local old_tags = s.tags
+    -- First try to get delete tags gently
+    for _, t in pairs(s.tags) do
+        t:delete()
+    end
+    -- Then be a little more fierce and force clients to go somewhere random
+    local fallback = nil
+    for other_screen in capi.screen do
+        if #other_screen.tags > 0 then
+            fallback = other_screen.tags[1]
```

Comment:
Oh, I totally forgot to say "I don't like this part at all and I'm open for suggestions".
How about some "special" signal just for "this tag must go away, do something suitably nice" that is emitted at the beginning of this function? That way tyrannical can hook into that signal and do everything it needs to do and the rest of this function won't be necessary.

Thanks for this idea!

So, what do you think about:
```
screen.connect_signal("removed", function(s)
    for _, t in pairs(s:tags) do
        t:emit_signal("delete-for-screen-removal")
    end
    [Now here the rest of the code]
end)
```
(Also: Ideas for a good name for the signal?)